subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplier.java

```diff
@@ -98,4 +102,36 @@ private String defaultMavenToolchainsDefinitionsLocation() {
         return new File(MavenUtil.getUserMavenDir(), "toolchains.xml").getAbsolutePath();
     }
 
+    /**
+     * Basic error handler which logs via SLF4J used to override
+     * default behavior which logs directly to stderr.
+     */
+    private static class LoggingErrorHandler implements ErrorHandler {
```

Comment:
Great to see we have an option for handling parsing errors.
However, since we will log the fact that we ignored the Maven toolchains in case of errors, I would suggest to collect those and add them to our own logged message instead of passing them through.

Reply:
Warnings and Errors do not propagate the exception -- only Fatal Errors. So, only Fatal Errors reach our catch block where the original log was, which thus causes us to log that the toolchain file failed to be read. 

I see a few options here:
1) Fail if we get any exception -- warnings, errors, fatal errors. Then our catch block will do the logging. 
2) Collect all exceptions, print them out at the end regardless of a fatal error/catch block getting reached.
3) Collect all exceptions, only print them if we reach a fatal error in our catch block.